### PR TITLE
Add `survey` action and Privacy Pro attributes

### PR DESCRIPTION
**Pull request by samsymons**
Task/Issue URL: https://app.asana.com/0/1193060753475688/1207234800675206/f
Tech Design URL:
CC:

**Description**:

Client PR for https://github.com/duckduckgo/BrowserServicesKit/pull/826

**Steps to test this PR**:
1. Update the app to use the staging URL from https://github.com/duckduckgo/remote-messaging-config/pull/58
2. Check that the message appears under the correct circumstances, e.g. you are signed into a Privacy Pro account
3. Check that the generated URL contains the parameters requested by the survey

**Device Testing**:

* [ ] iPhone SE (1st Gen)
* [ ] iPhone 8
* [ ] iPhone X
* [ ] iPhone 14 Pro
* [ ] iPad

**OS Testing**:

* [ ] iOS 14
* [ ] iOS 15
* [ ] iOS 16

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: DuckDuckGo/RemoteMessagingSurveyURLBuilder.swift [hunk 1]
```diff
@@ -83,6 +74,8 @@ struct DefaultSurveyURLBuilder: SurveyURLBuilder {
                 if let daysSinceLastActive = activationDateStore.daysSinceLastActive() {
                     queryItems.append(URLQueryItem(name: parameter.rawValue, value: String(describing: daysSinceLastActive)))
                 }
+            case .daysInstalled:
+                break // TODO
```
Comment (amddg44) [inline DuckDuckGo/RemoteMessagingSurveyURLBuilder.swift]: `.daysInstalled` is `TODO` 🙂

File: DuckDuckGo/RemoteMessagingSurveyURLBuilder.swift [hunk 2]
```diff
@@ -17,18 +17,13 @@
...
 import Common
 
-protocol SurveyURLBuilder {
-    func addSurveyParameters(to url: URL) -> URL
-}
-
-struct DefaultSurveyURLBuilder: SurveyURLBuilder {
+struct DefaultRemoteMessagingSurveyURLBuilder: RemoteMessagingSurveyActionMapping {
 
     enum SurveyURLParameters: String, CaseIterable {
```
Comment (amddg44) [inline DuckDuckGo/RemoteMessagingSurveyURLBuilder.swift]: I think this enum can be safely deleted?

Comment (samsymons) [reply DuckDuckGo/RemoteMessagingSurveyURLBuilder.swift]: Yes indeed, good catch - thanks!

File: DuckDuckGo/RemoteMessagingSurveyURLBuilder.swift (hunk 1 above)
Comment (samsymons) [reply DuckDuckGo/RemoteMessagingSurveyURLBuilder.swift]: Fixed!

**Review by amddg44 — APPROVED** [review]
LGTM! Tested thoroughly & all worked exactly as expected

**Pull request merged**